### PR TITLE
feat: worker auth endpoints

### PR DESCRIPTION
**Pull request by plyr4**
this adds the server functionality introduced in https://github.com/go-vela/server/pull/790
it also cleans up a couple tests

### CODE REVIEW

**Review by ecrupper — APPROVED** [review]
LGTM

File: vela/worker_test.go [hunk 1]
```diff
@@ -300,6 +351,24 @@ func ExampleWorkerService_Add() {
 	fmt.Printf("Received response code %d, for worker %+v", resp.StatusCode, worker)
 }
 
+func ExampleWorkerService_RefreshAuth() {
```
Comment (cognifloyd) [inline vela/worker_test.go:354]: Where is this used? Same goes for the rest of the `ExampleWorkerService_*` funcs.

**Review by cognifloyd — APPROVED** [review]
LGTM. I have one question about why we need some functions, but that doesn't block merging this.

File: vela/worker_test.go (hunk 1 above)
Comment (plyr4) [reply vela/worker_test.go:354]: honestly im not sure. i added it for consistency

Comment (plyr4) [reply vela/worker_test.go:354]: seemed like leftover boiler plate tests from the initial worker router commits. we can probably remove them in favor the more details tests above them

Comment (wass3rw3rk) [reply vela/worker_test.go:354]: this is used for documentation, see https://pkg.go.dev/github.com/go-vela/sdk-go@v0.18.1/vela#example-WorkerService.Update as an example which is extracted from `ExampleWorkerService_Update`

**Review by wass3rw3rk — APPROVED** [review]

File: vela/worker_test.go (hunk 1 above)
Comment (plyr4) [reply vela/worker_test.go:354]: oh!!! 😮‍💨 thats awesome. i didnt know go doc used test naming to populate that. thanks @wass3rw3rk

**Pull request merged**